src/linker/Linker/LinkContext.cs

```diff
@@ -463,10 +465,20 @@ public bool IsOptimizationEnabled (CodeOptimizations optimization, AssemblyDefin
 			return Optimizations.IsEnabled (optimization, context);
 		}
 
-		public void LogMessage (MessageContainer message)
+		public void LogMessage (MessageContainer? message)
```

Comment:
That's not a great API. You could experiment with static MessageContainer.Empty sentinel.